src/dmd/cppmangle.d

```diff
@@ -1259,58 +1261,45 @@ extern(C++):
          * Ds       char16_t
          * u <source-name>  # vendor extended type
          */
-        char c;
-        char p = 0;
+
+        // Handle any target-specific basic types.
+        if (auto tm = target.cppTypeMangle(t))
+        {
+            bool builtin = (t.ty == Tvoid || t.ty == Tbool || t.isintegral() || t.isreal());
```

Comment:
A change in API, but I think it's better to leave the decision whether to substitute or not with `target.cppTypeMangle`, e.g. by passing it a ref bool. Or at least the possibility to override it:
```d
bool builtin = (t.ty == Tvoid || t.ty == Tbool || t.isintegral() || t.isreal());
if (auto tm = target.cppTypeMangle(t, builtin)) // could write to builtin
{
             
```

Reply:
C++ abi doesn't substitute what it calls fundamental types. This is a constant that isn't target specific.

Reply:
Would be nice if you could add that as a comment to the code. (also using `isFundamentalType` variable instead of `builtin`)
Indeed: `std::nullptr_t` is also a fundamental type, and is correctly not substituted.

Reply:
Renamed both this and the writeBasicType parameter to `isFundamentalType`.